### ISSUE TITLE
System tray title can be too long on Windows

### ISSUE THREAD

**Issue by TheLovinator1** [Windows200000/TwitchDropsMiner-updated#87]
The system tray title can be too long, causing a `ValueError: string too long (x, maximum length 128)`.

## Code Reference

https://github.com/Windows200000/TwitchDropsMiner-updated/blob/c9cd4adadaf0357567f4d10885cd05421560ecad/gui.py#L1048-L1057

## Suggested Solution

Truncate the title to 128 characters. Possible remove self.TITLE (`TITLE = "Twitch Drops Miner"`) from the title?
Or add an try-except block to catch the error and just skip updating the title.

## Example

Example system tray title (This is for a working drop, I have no idea what the title is for the drop that causes the error):
```
Twitch Drop Miner
Halo Infinite
Adrenal M40 AR 6.1% (0/3)
```

## Stack Trace

<details>
<summary>Stack trace (click to expand)</summary>
<br>

```python
10:36:26: Fatal error encountered:
10:36:26: 
10:36:26: Traceback (most recent call last):
10:36:26:   File "C:\Code\TwitchDropsMiner_2024-5-fix\main.py", line 158, in main
10:36:26:     await client.run()
10:36:26:   File "C:\Code\TwitchDropsMiner_2024-5-fix\twitch.py", line 769, in run
10:36:26:     await self._run()
10:36:26:   File "C:\Code\TwitchDropsMiner_2024-5-fix\twitch.py", line 982, in _run
10:36:26:     active_drop.display(countdown=False, subone=True)
10:36:26:   File "C:\Code\TwitchDropsMiner_2024-5-fix\inventory.py", line 242, in display
10:36:26:     self._manager.display_drop(self, countdown=countdown, subone=subone)
10:36:26:   File "C:\Code\TwitchDropsMiner_2024-5-fix\gui.py", line 2149, in display_drop
10:36:26:     self.tray.update_title(drop)  # tray
10:36:26:     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
10:36:26:   File "C:\Code\TwitchDropsMiner_2024-5-fix\gui.py", line 1115, in update_title
10:36:26:     self.icon.title = self.get_title(drop)
10:36:26:     ^^^^^^^^^^^^^^^
10:36:26:   File "C:\Code\TwitchDropsMiner_2024-5-fix\env\Lib\site-packages\pystray\_base.py", line 150, in title
10:36:26:     self._update_title()
10:36:26:   File "C:\Code\TwitchDropsMiner_2024-5-fix\env\Lib\site-packages\pystray\_win32.py", line 81, in _update_title
10:36:26:     self._message(
10:36:26:   File "C:\Code\TwitchDropsMiner_2024-5-fix\env\Lib\site-packages\pystray\_win32.py", line 337, in _message
10:36:26:     win32.Shell_NotifyIcon(code, win32.NOTIFYICONDATAW(
10:36:26:                                  ^^^^^^^^^^^^^^^^^^^^^^
10:36:26: ValueError: string too long (131, maximum length 128)
10:36:26: 
10:36:26: Exiting...
10:36:26: 
10:36:26: Application Terminated.
10:36:26: Close the window to exit the application.
```
</details>

## Additional Information

@woctezuma made an issue about this in the original repository:
https://github.com/DevilXD/TwitchDropsMiner/issues/460

<details>
<summary>Response from DevilXD (click to expand)</summary>
<br>

```markdown	
The error going away doesn't resolve the underlying issue, which in this case appears to be 
a 128 characters limit on a tray icon title. This can apparently be exceeded, if a drop 
that's currently being mined has a long name, or belongs to a game that has a long name, 
or a combination of these two.

The simplest solution would be to just always trim the tray icon title down to <128 characters, 
but that'd just make it ugly and weirdly cut-off in these cases. A proper solution would be to always
leave the main title and progress percentage up, while only trimming the drop's name. 
If that's isn't enough, then trim the game's name as well. Maybe the application title could be removed as well,
solely to allow for more characters to be displayed. I'll try to put out a patch for this soon.
```

</details>

**Comment by Windows200000** [comment]
> Truncate the title to 128 characters. Possible remove self.TITLE (`TITLE = "Twitch Drops Miner"`) from the title?
> Or add an try-except block to catch the error and just skip updating the title.

As @DevilXD said, the drop name should be trimmed. Should be fairly easy to do.

**Comment by Windows200000** [comment]
Fixed in https://github.com/Windows200000/TwitchDropsMiner-updated/commit/1c1a950ab4f5f06e79e71e58efae9f6f58e9cb88

https://github.com/Windows200000/TwitchDropsMiner-updated/blob/1c1a950ab4f5f06e79e71e58efae9f6f58e9cb88/gui.py#L1048-L1079

**Comment by TheLovinator1** [comment]
```
2024-06-01 16:31:06: Progress: 1/240 - Campaign(Vampire: The Masquerade - Bloodhunt, Bloodhunt June Drops, 0/1)
len after: 128
Twitch Drops Miner
Vampire: The Masquerade - Bloodhunt
Kill Time, Scribbled Away, Unworship, Burnout, Milk, Swift Vi… 0.4% (0/1)
```

Thanks, it works

**Comment by Windows200000** [comment]
> ![image](https://private-user-images.githubusercontent.com/4153203/335807922-051525db-10c3-4560-ab31-3bc78e66a02f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTcyNTUyNzEsIm5iZiI6MTcxNzI1NDk3MSwicGF0aCI6Ii80MTUzMjAzLzMzNTgwNzkyMi0wNTE1MjVkYi0xMGMzLTQ1NjAtYWIzMS0zYmM3OGU2NmEwMmYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MDFUMTUxNjExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjFlZjA1MzkyNWU1YzY0ZGJhNmU4ZGJiYmIxMzkzMjEzNzI1MzllNWRjNzI4MTNiZGY4MzQyNTg0MDE5YzFiMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.6XDvZVpztzgPfhViCeewOv15CJw7acpz_gLml2XxFQc)
> 
> ```
> 2024-06-01 16:31:06: Progress: 1/240 - Campaign(Vampire: The Masquerade - Bloodhunt, Bloodhunt June Drops, 0/1)
> len after: 128
> Twitch Drops Miner
> Vampire: The Masquerade - Bloodhunt
> Kill Time, Scribbled Away, Unworship, Burnout, Milk, Swift Vi… 0.4% (0/1)
> ```
> 
> Thanks, it works

Interesting, well first of all I forgot to remove the print statement, but also it allows 128 but only prints 127. Will fix that.

**Comment by Windows200000** [comment]
Fixed in [v15.6.0](https://github.com/Windows200000/TwitchDropsMiner-updated/releases/tag/v15.6.0)
